volatility3/cli/volshell/generic.py

```diff
@@ -418,67 +445,194 @@ def display_type(
+            # attempt to follow null pointers.
+            if volobject.is_readable() and volobject != 0:
+                # now deference the pointer and store this as the new volobject
+                volobject = volobject.dereference()
+                dereference_count = dereference_count + 1
+            else:
+                # if we aren't able to follow the pointers anymore then there will
+                # be no more information to display as we've already printed the
+                # details of this pointer including the fact that we're not able to
+                # follow it anywhere
+                return
 
         if hasattr(volobject.vol, "members"):
+            # display the header for this object, if the orginal object was just a type string, display the type information
+            struct_header = f'{"    " * dereference_count}{volobject.vol.type_name} ({volobject.vol.size} bytes)'
+            if isinstance(object, str) and offset is None:
+                suffix = ":"
+            else:
+                # this is an actual object or an offset was given so the offset should be displayed
+                suffix = f" @ {hex(volobject.vol.offset)}:"
+            print(struct_header + suffix)
+
+            # it is a more complex type, so all members also need information displayed
             longest_member = longest_offset = longest_typename = 0
             for member in volobject.vol.members:
                 relative_offset, member_type = volobject.vol.members[member]
                 longest_member = max(len(member), longest_member)
                 longest_offset = max(len(hex(relative_offset)), longest_offset)
-                longest_typename = max(len(member_type.vol.type_name), longest_typename)
+                member_type_name = self._get_type_name_with_pointer(
+                    member_type
+                )  # special case for pointers to show what they point to
+                longest_typename = max(len(member_type_name), longest_typename)
+                if longest_typename > MAX_TYPENAME_DISPLAY_LENGTH:
+                    longest_typename = MAX_TYPENAME_DISPLAY_LENGTH
 
             for member in sorted(
                 volobject.vol.members, key=lambda x: (volobject.vol.members[x][0], x)
             ):
                 relative_offset, member_type = volobject.vol.members[member]
                 len_offset = len(hex(relative_offset))
                 len_member = len(member)
-                len_typename = len(member_type.vol.type_name)
+                member_type_name = self._get_type_name_with_pointer(
+                    member_type
+                )  # special case for pointers to show what they point to
+                if len(member_type_name) > MAX_TYPENAME_DISPLAY_LENGTH:
+                    member_type_name = (
+                        f"{member_type_name[:MAX_TYPENAME_DISPLAY_LENGTH - 3]}..."
+                    )
+                len_typename = len(member_type_name)
```

Comment:
Feels marginally tidier, but not a big deal...
```suggestion
                len_typename = len(member_type_name)
                if len(member_type_name) > MAX_TYPENAME_DISPLAY_LENGTH:
                    len_typename = MAX_TYPENAME_DISPLAY_LENGTH
                    member_type_name = f"{member_type_name[:len_typename - 3]}..."
```